src/Sulu/Bundle/MarkupBundle/Markup/HtmlTagExtractor.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\MarkupBundle\Markup;
+
+/**
+ * Extracts tags from html content.
+ */
+class HtmlTagExtractor implements TagExtractorInterface
+{
+    const COUNT_REGEX = '/<%1$s:[a-z]+/';
+    const ATTRIBUTE_REGEX = '/(?<name>\b[\w-]+\b)\s*=\s*"(?<value>[^"]*)"/';
+    const CONTENT_REGEX = '/(?:>(?<content>[^<]*)<)/';
+    const TAG_REGEX = '/(?<tag><%1$s:(?<name>[a-z]+)((?!>|\/>).)*(?:\/>|>((?!<\/%1$s:\2*>).)*<\/%1$s:\2*>))/';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count($html, $namespace)
```

Comment:
Do we really need that `count` method? From what I see it is only called once for a simple check, and the `extract` method is called directly after that check. Can't we use the result of the `extract` method for that, instead of introducing another method?

The same regex is executed twice, which results in worse performance, and it makes this interface more complex.